### PR TITLE
feat(kuma-cp): added docs for tls prom secured

### PR DESCRIPTION
**Pull request by lukidzi**
<Explain your change!>

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits)

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: .github/styles/Vocab/Base/accept.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[Kk]ong
+ [Kk]ong
```
Comment (slonka) [inline .github/styles/Vocab/Base/accept.txt:1]: Seems not relevant

```suggestion
[Kk]ong
```

File: app/_src/explore/observability.md [hunk 2]
```diff
@@ -46,6 +46,9 @@ There are 2 ways you can run prometheus:
 
 1. Inside the mesh (default for [`kumactl install observability`](#demo-setup)). In this case you can use mTLS to retrieve the metrics. This provides high security but will require one prometheus per mesh and might not be accessible if your mesh becomes unavailable. It will also require one Prometheus deployment per {{site.mesh_product_name}} mesh.
 2. Outside the mesh. In this case you'll need to specify `skipMTLS: true` in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics). This is less secured but will ensure Prometheus is as available as possible. It is also easier to add to an existing setup with services in and outside the mesh.
+{% if_version lte:2.3.x %}
+3. Outside the mesh with TLS enabled. In this case you'll need to provide certificates for each dataplane and specify configuration in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics#secure-metrics-with-tls). This is more secured than second option but requires more configuration.
```
Comment (slonka) [inline app/_src/explore/observability.md:50]:
```suggestion
3. Outside the mesh with TLS enabled. In this case you'll need to provide certificates for each dataplane and specify configuration in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics#secure-metrics-with-tls). This is more secure than second option but requires more configuration.
```

File: app/_src/policies/traffic-metrics.md [hunk 3]
```diff
@@ -100,6 +112,88 @@ The metrics endpoint is forwarded to the standard Envoy [Prometheus metrics endp
 You can pass the `filter` query parameter to limit the results to metrics whose names match a given regular expression.
 By default all available metrics are returned.
 
+{% if_version lte:2.3.x %}
+### Secure metrics with TLS
+
+{{site.mesh_product_name}} allows configuring metrics endpoint with TLS. You can use it when the `Prometheus` deployment is outside of the mesh and requires security for the communication.
```
Comment (slonka) [inline app/_src/policies/traffic-metrics.md:118]:
```suggestion
{{site.mesh_product_name}} allows configuring metrics endpoint with TLS. You can use it when the `Prometheus` deployment is outside of the mesh and requires secure communication.
```

File: app/_src/policies/traffic-metrics.md [hunk 4]
```diff
@@ -100,6 +112,88 @@ The metrics endpoint is forwarded to the standard Envoy [Prometheus metrics endp
...
+    backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        port: 5670
+        path: /metrics
+        tls:
+          mode: delegated
+```
+Apart from `Mesh` configuration, `kuma-sidecar` requires provided certificate and key to the `kuma-sidecar`. When the certificate and the key is available in the container, `kuma-sidecar` needs paths to them provided in environment variables:
```
Comment (slonka) [inline app/_src/policies/traffic-metrics.md:140]:
```suggestion
In addition to the `Mesh` configuration, `kuma-sidecar` requires a provided certificate and key for its operation. When the certificate and key are available within the container, `kuma-sidecar` needs the paths to these files to be provided as the following environment variables:
```

File: app/_src/policies/traffic-metrics.md [hunk 5]
```diff
@@ -100,6 +112,88 @@ The metrics endpoint is forwarded to the standard Envoy [Prometheus metrics endp
...
+
+* KUMA_DATAPLANE_METRICS_CERT_PATH
+* KUMA_DATAPLANE_METRICS_KEY_PATH
+
+It's possible to use [`ContainerPatch`](/docs/{{ page.version }}/production/dp-config/dpp-on-kubernetes/#custom-container-configuration) to add variable into `kuma-sidecar`:
```
Comment (slonka) [inline app/_src/policies/traffic-metrics.md:145]:
```suggestion
It's possible to use a [`ContainerPatch`](/docs/{{ page.version }}/production/dp-config/dpp-on-kubernetes/#custom-container-configuration) to add variables to `kuma-sidecar`:
```

File: app/_src/policies/traffic-metrics.md [hunk 6]
```diff
@@ -100,6 +112,88 @@ The metrics endpoint is forwarded to the standard Envoy [Prometheus metrics endp
...
+      path: /metrics
+      tls:
+        mode: delegated
+```
+
+Apart from `Mesh` configuration, `kuma-dp` requires provided certificate and key to the `kuma-dp`. Upload the certificate and the key to the machine and define:
+* KUMA_DATAPLANE_METRICS_CERT_PATH
+* KUMA_DATAPLANE_METRICS_KEY_PATH
+
+for `kuma-dp` with paths to them.
```
Comment (slonka) [inline app/_src/policies/traffic-metrics.md:191]:
```suggestion
In addition to the `Mesh` configuration, `kuma-dp` requires a provided certificate and key for its operation. Please upload the certificate and the key to the machine, and then define the following environment variables with the correct paths:
* KUMA_DATAPLANE_METRICS_CERT_PATH
* KUMA_DATAPLANE_METRICS_KEY_PATH
```

**Review by slonka — APPROVED** [review]
Mostly nitpicks

File: app/_src/policies/traffic-metrics.md [hunk 7]
```diff
@@ -72,6 +76,10 @@ metrics:
     type: prometheus
     conf:
       skipMTLS: true # by default mTLS metrics are also protected by mTLS. Scraping metrics with mTLS without transparent proxy is not supported at the moment.
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:78]: Only put this if lte:2.3.x right?

File: app/_src/policies/traffic-metrics.md [hunk 8]
```diff
@@ -72,6 +76,10 @@ metrics:
...
+{% if_version lte:2.3.x %}
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:79]:
```suggestion
{% if_version gte:2.4.x %}
```

right?

File: app/_src/policies/traffic-metrics.md [hunk 9]
```diff
@@ -90,6 +98,10 @@ metrics:
       path: /metrics
       tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS  
         kuma.io/service: dataplane-metrics
+{% if_version lte:2.3.x %}
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:101]:
```suggestion
{% if_version gte:2.3.x %}
```

File: app/_src/policies/traffic-metrics.md [hunk 10]
```diff
@@ -100,6 +112,88 @@ The metrics endpoint is forwarded to the standard Envoy [Prometheus metrics endp
 You can pass the `filter` query parameter to limit the results to metrics whose names match a given regular expression.
 By default all available metrics are returned.
 
+{% if_version lte:2.3.x %}
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:115]:
```suggestion
{% if_version gte:2.4.x %}
```

**Review by lahabana — CHANGES_REQUESTED** [review]
This change should conditionally change all the places we refer to `skipMTLS` to recommend the new option as this is what we want to go with in the future right?

**Comment by lukidzi** [comment]
> This change should conditionally change all the places we refer to `skipMTLS` to recommend the new option as this is what we want to go with in the future right?

Yes, I will apply the change to the docs, to recommend a new model since 2.4

File: app/_src/explore/observability.md [hunk 11]
```diff
@@ -45,7 +45,10 @@ This service discovery will connect to the control plane and retrieve all data p
 There are 2 ways you can run prometheus:
 
 1. Inside the mesh (default for [`kumactl install observability`](#demo-setup)). In this case you can use mTLS to retrieve the metrics. This provides high security but will require one prometheus per mesh and might not be accessible if your mesh becomes unavailable. It will also require one Prometheus deployment per {{site.mesh_product_name}} mesh.
-2. Outside the mesh. In this case you'll need to specify `skipMTLS: true` in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics). This is less secured but will ensure Prometheus is as available as possible. It is also easier to add to an existing setup with services in and outside the mesh.
+2. Outside the mesh. In this case you'll need to specify {% if_version lte:2.3.x %}`skipMTLS: true`{% endif_version %}{% if_version gte:2.4.x %}`tls.mode: disabled`{% endif_version %} in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics). This is less secured but ensures Prometheus is as available as possible. It's also easier to add to an existing setup with services in and outside the mesh.
```
Comment (lahabana) [inline app/_src/explore/observability.md:48]:
```suggestion
2. Outside the mesh. In this case you'll need to specify {% if_version lte:2.3.x %}`skipMTLS: true`{% endif_version %}{% if_version gte:2.4.x %}`tls.mode: disabled`{% endif_version %} in the [traffic metrics configuration](/docs/{{ page.version }}/policies/traffic-metrics). This is less secure but ensures Prometheus is as available as possible. It's also easier to add to an existing setup with services in and outside the mesh.
```

File: app/_src/policies/traffic-metrics.md [hunk 12]
```diff
@@ -110,14 +251,15 @@ Here are reasons where you'd want to use this feature:
 
 - Application metrics are labelled with your mesh parameters (tags, mesh, data plane name...), this means that in mixed Universal and Kubernetes mode metrics are reported with the same types of labels.
 - Both application and sidecar metrics are scraped at the same time. This makes sure they are coherent (with 2 different scrapers they can end up scraping at different intervals and make metrics harder to correlate).
-- If you disable [passthrough](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing) and your mesh uses mTLS but Prometheus is outside the mesh (`skipMTLS: true`) this will be the only way to retrieve these metrics as the application is completely hidden behind the sidecar. 
+- If you disable [passthrough](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing) and your mesh uses mTLS but Prometheus is outside the mesh {% if_version lte:2.3.x %}(`skipMTLS: true`){% endif_version %}{% if_version gte:2.4.x %}`tls.mode: disabled`{% endif_version %} this is the only way to retrieve these metrics as the app is completely hidden behind the sidecar.
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:254]:
```suggestion
- If you disable [passthrough](/docs/{{ page.version }}/networking/non-mesh-traffic#outgoing) and your mesh uses mTLS but Prometheus is outside the mesh {% if_version lte:2.3.x %}(`skipMTLS: true`){% endif_version %}{% if_version gte:2.4.x %}(`tls.mode: disabled`){% endif_version %} this is the only way to retrieve these metrics as the app is completely hidden behind the sidecar.
```

File: app/_src/policies/traffic-metrics.md [hunk 13]
```diff
@@ -135,6 +277,10 @@ spec:
         path: /metrics
         tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
           kuma.io/service: dataplane-metrics
+{% if_version gte:2.4.x %}
+        tls:
+          mode: activeMTLSBackend
+{% endif_version %}
```
Comment (lahabana) [inline app/_src/policies/traffic-metrics.md:283]: I think this is left over

**Review by lahabana — APPROVED** [review]
LGTM! modulo the nits

**Pull request merged**